services/registry/middleware/bearer-auth.js

```diff
@@ -1,26 +1,15 @@
 'use strict';
 
-const Token = require('../models/token');
 const { response } = require('boltzmann');
+const crypto = require('crypto');
 
 module.exports = createBearerAuthMW;
 
-const idempotent = new Set(['GET', 'HEAD']);
-
-function createBearerAuthMW() {
+function createBearerAuthMW({
+  sessionTimeout = Number(process.env.SESSION_TIMEOUT) || 5 * 60
+} = {}) {
```

Comment:
The registry layer _always_ attempts to trade the `Authorization` header for a `User` object when given. It'll try redis first, then fall back to asking the storage layer.